subprojects/dependency-management/src/integTest/groovy/org/gradle/integtests/resolve/StronglyTypedConfigurationAttributesResolveIntegrationTest.groovy

```diff
@@ -363,4 +369,153 @@ class StronglyTypedConfigurationAttributesResolveIntegrationTest extends Abstrac
         executedAndNotSkipped ':b:foo2Jar'
         notExecuted ':b:fooJar', ':b:barJar', ':b:bar2Jar'
     }
+
+    def "can select configuration thanks to producer schema disambiguation"() {
+        given:
+        file('settings.gradle') << "include 'a', 'b'"
+        buildFile << """
+            $typeDefs
+
+            project(':b') {
+               configurationAttributesSchema {
+                  matchStrictly(buildType)
+                  setMatchingStrategy(flavor, new AttributeMatchingStrategy<String>() {
+                       boolean isCompatible(String req, String can) { req == can }
+                       public <K> List<K> selectClosestMatch(AttributeValue<String> value, Map<K, String> compatibleValues) {
```

Comment:
Good catch, I think the variable needs a rename. It's leftover from the previous story where only compatible values were passed. But now, they are either compatible or provided. I will rename to `candidateValues`